### PR TITLE
Mulqacc iss fix

### PR DESCRIPTION
**Pull request by GregAC**

### CODE REVIEW

File: hw/ip/otbn/dv/otbnsim/sim/insn.py [hunk 1]
```diff
@@ -148,10 +148,12 @@ def execute(self, model: OTBNModel) -> None:
 
         if self.wb_variant > 0:
             if self.wb_variant == 1:
-                model.set_wr_halfword(self.wrd, acc, self.wrd_hwsel)
-                acc = acc >> 128
-            elif self.wb_variant == 2:
                 model.state.wreg[self.wrd].set(acc)
+            elif self.wb_variant >= 2:
+                # set_wr_halfword expects 0 in upper 128 bits
+                acc_lower = acc & 0xFFFF_FFFF_FFFF_FFFF_FFFF_FFFF_FFFF_FFFF
```
Comment (rswarbrick) [inline hw/ip/otbn/dv/otbnsim/sim/insn.py:154]: Could you write the mask as `(1 << 128) - 1`? Counting F's gets old pretty fast.

File: hw/ip/otbn/dv/otbnsim/sim/isa.py [hunk 2]
```diff
@@ -291,11 +291,11 @@ class InstructionBNAQType(InstructionW3Type):
 
 To specify, use the literal syntax .Z""")
     field_wrd_hwsel = Field(name="wrd_hwsel", base=29, size=1, description="")
-    field_wrs1_qwsel = Field(name="wrs1_qwsel",
+    field_wrs2_qwsel = Field(name="wrs2_qwsel",
                              base=27,
                              size=2,
                              description="")
-    field_wrs2_qwsel = Field(name="wrs2_qwsel",
+    field_wrs1_qwsel = Field(name="wrs1_qwsel",
```
Comment (rswarbrick) [inline hw/ip/otbn/dv/otbnsim/sim/isa.py:298]: Could you swap the ordering here? Now we've got `wrs2_qwsel` before `wrs1_qwsel`.

**Review by rswarbrick — CHANGES_REQUESTED** [review]

File: hw/ip/otbn/dv/otbnsim/sim/isa.py (hunk 2 above)
Comment (GregAC) [reply hw/ip/otbn/dv/otbnsim/sim/isa.py:298]: I left it that way as it matches the field ordering going from highest bit to lowest bit (so `wrd_hwsel` at 29 then `wrs2_qwsel` at 28-27 and `wrs1_qwsel` at 26-25)

Comment (rswarbrick) [reply hw/ip/otbn/dv/otbnsim/sim/isa.py:298]: Oh, good point.

**Review by rswarbrick — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**